src/main/java/org/generationcp/middleware/api/germplasm/GermplasmServiceImpl.java

```diff
@@ -1223,6 +1224,26 @@ public ProgenitorsDetailsDto getGermplasmProgenitorDetails(final Integer gid) {
 		return null;
 	}
 
+	@Override
+	public void updateGermplasmBasicDetails(final Integer gid, final GermplasmBasicDetailsDto germplasmBasicDetailsDto) {
+		final Germplasm germplasm = this.daoFactory.getGermplasmDao().getById(gid);
+
+		final Optional<Integer> locationIdOptional =
```

Comment:
You can replace this conditional using: Optional.ofNullable(germplasmBasicDetailsDto.getBreedingLocationId()).ifPresent(germplasm::setLocationId);